controllers/appgroup_controller.go

```diff
@@ -503,3 +509,21 @@ func (r *ApplicationGroupReconciler) generateReverseWorkflow(ctx context.Context
 	}
 	return nil
 }
+
+// suspend a workflow if it is not already finished or suspended
+func (r *ApplicationGroupReconciler) suspendWorkflow(ctx context.Context, logr logr.Logger, wf *v1alpha12.Workflow) error {
```

Comment:
Should we also delete it? Or just suspend it?

Reply:
Can we also add a short test case for this scenario?

Reply:
> Should we also delete it? Or just suspend it?

Deleting the forward workflow will cause all HelmReleases to be garbage collected (I think)

Reply:
We don't set an ownership reference on the HelmReleases so why would this heppen?

Reply:
I have noticed when we force delete an application group (by removing the finalizer) all associated HelmReleases are cleaned up. Is that not the case?

Reply:
BTW we are adding the FW workflow as an owner of the Reverse WF
```go
			// Add OwnershipReference
			err = controllerutil.SetControllerReference(wf, a.rwf, a.scheme)
			if err != nil {
				l.Error(err, "unable to set forward workflow as owner of Argo reverse Workflow object")
				return fmt.Errorf("unable to set forward workflow as owner of Argo reverse Workflow: %w", err)
			}
```

Reply:
> We don't set an ownership reference on the HelmReleases so why would this happen?
You're right this doesn't happen. I suppose it would be ok to delete the FW workflow provided we remove the Forward workflow as the owner of the reverse WF

Reply:
Added a test case for suspend